File: SubscriptionApi/SubscriptionApi/Commands/ExceptionLoggedCommand.cs
using System;

namespace SubscriptionApi.Commands
{
  public class ExceptionLoggedCommand : Command
  {
    private readonly Log _log;
    private readonly Command _wrappedCommand;

    public ExceptionLoggedCommand(Log log, Command wrappedCommand)
    {
      _log = log;
      _wrappedCommand = wrappedCommand;
    }

    public void Invoke()
    {
      try
      {
        _wrappedCommand.Invoke();
      }
      catch (Exception e)
      {
        _log.Error(e);
      }
    }
  }
}

File: Java/factories/_01_no_factories/src/main/java/Commands/ExceptionLoggedCommand.java
package Commands;

import other.Log;

public class ExceptionLoggedCommand implements Command {
    private final Log _log;
    private final Command _wrappedCommand;

    public ExceptionLoggedCommand(Log log, Command wrappedCommand) {
        _log = log;
        _wrappedCommand = wrappedCommand;
    }

    public void Invoke() {
        try {
            _wrappedCommand.Invoke();
        } catch (Exception e) {
            _log.Error(e);
        }
    }
}
